app/controllers/usps_upload_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UspsUploadController < ApplicationController
+  def create
+    authorize do
+      UspsUploader.new.run unless HolidayService.observed_holiday?(today)
+      render plain: 'ok'
+    end
+  end
+
+  private
+
+  def authorize
+    # Check for empty to make sure that the token is configured
+    if authorization_token && authorization_token == Figaro.env.usps_upload_token
+      yield
+    else
+      head :unauthorized
+    end
+  end
+
+  def authorization_token
+    request.headers['X-USPS-UPLOAD-TOKEN']
```

Comment:
This is fine. Another option that we were considering for the twilio endpoint was to use basic auth. It's essentially the same thing but doesn't require a special (`X-`) header. The only real difference is in this line.